core/src/main/java/com/novoda/merlin/receiver/ConnectivityCallbacks.java

```diff
@@ -0,0 +1,50 @@
+package com.novoda.merlin.receiver;
+
+import android.annotation.TargetApi;
+import android.net.ConnectivityManager;
+import android.net.Network;
+import android.net.NetworkInfo;
+import android.os.Build;
+
+import com.novoda.merlin.service.MerlinService;
+
+@TargetApi(Build.VERSION_CODES.LOLLIPOP)
+public class ConnectivityCallbacks extends ConnectivityManager.NetworkCallback {
+
+    private final ConnectivityManager connectivityManager;
+    private final MerlinService merlinService;
```

Comment:
Hi @tasomaniac 😄 , thanks for reviewing! 

As far as the `MerlinService` goes. It is meant to be a purely `BoundService` from what I have seen. We've made sure to drop all collaborators in the event that the service is `unbound` is there still an instance where the `Service` could be killed but the `unbind` method not called? If so then I'll create a task to investigate and tackle this because it is likely a bug 😰